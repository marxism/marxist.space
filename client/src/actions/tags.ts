import { fetchTags as _fetchTags } from '../util/api'

export const FETCH_TAGS_REQUEST = 'FETCH_TAGS_REQUEST'
export const FETCH_TAGS_SUCCESS = 'FETCH_TAGS_SUCCESS'
export const FETCH_TAGS_ERROR = 'FETCH_TAGS_ERROR'

const fetchTagsRequest = { type: FETCH_TAGS_REQUEST }
const fetchTagsSuccess = (tags) => ({ type: FETCH_TAGS_SUCCESS, tags })
const fetchTagsError = (error) => ({ type: FETCH_TAGS_ERROR, error })

export const fetchTags = () => async (dispatch) => {
  dispatch(fetchTagsRequest)
  try {
    const tags = await _fetchTags()
    dispatch(fetchTagsSuccess(tags))
  } catch (error) {
    dispatch(fetchTagsError(error))
  }
}
