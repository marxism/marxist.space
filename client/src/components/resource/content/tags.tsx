import React from 'react'
import styled from 'styled-components/macro'

const Wrapper = styled.div`
  font-size: 12px;
  margin-top: auto;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  & > * {
    margin-right: 4px;
  }

  & > span {
    color: ${(props) => props.theme.mutedText};
  }
`

type Props = {
  tags: string[]
}

const Tags = (props: Props) => (
  <Wrapper>
    <span>Tags: {props.tags.join(' ')}</span>
  </Wrapper>
)

export default Tags
