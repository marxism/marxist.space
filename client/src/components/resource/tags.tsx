import * as React from 'react'
import styled from 'styled-components/macro'

const Span = styled.span`
  font-size: 12px;
  margin-top: auto;
  color: ${(props) => props.theme.mutedText};
`

type Props = {
  tags: Array<string>
}

const Tags = (props: Props) => <Span>Tags: {props.tags.join(' ')}</Span>

export default Tags
