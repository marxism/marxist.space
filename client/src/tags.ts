import { getTags } from './util/api'

// const tags = getTags()

const tags = ['one', 'two']

export default tags
