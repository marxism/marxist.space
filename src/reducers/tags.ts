import {
  FETCH_TAGS_REQUEST,
  FETCH_TAGS_SUCCESS,
  FETCH_TAGS_ERROR,
} from '../actions/tags'

const initialState = { isFetching: false, items: [] }

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TAGS_REQUEST:
      return { ...state, isFetching: true, tags: null }
    case FETCH_TAGS_SUCCESS:
      return { ...state, isFetching: false, items: action.tags }
    case FETCH_TAGS_ERROR:
      return { ...state, isFetching: false }

    default:
      return state
  }
}
