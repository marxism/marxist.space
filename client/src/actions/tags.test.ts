/* eslint-env jest */

import * as tagsActions from './tags'

describe('tags actions', () => {
  test('fetchTagsRequest', () => {
    expect(tagsActions.fetchTagsRequest).toMatchSnapshot()
  })

  test('fetchTagsSuccess', () => {
    expect(tagsActions.fetchTagsSuccess(['one', 'two'])).toMatchSnapshot()
  })

  test('fetchTagsError', () => {
    expect(tagsActions.fetchTagsError('oh no')).toMatchSnapshot()
  })
})
