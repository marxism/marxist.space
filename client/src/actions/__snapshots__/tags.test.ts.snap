// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags actions fetchTagsError 1`] = `
Object {
  "error": "oh no",
  "type": "FETCH_TAGS_ERROR",
}
`;

exports[`tags actions fetchTagsRequest 1`] = `
Object {
  "type": "FETCH_TAGS_REQUEST",
}
`;

exports[`tags actions fetchTagsSuccess 1`] = `
Object {
  "tags": Array [
    "one",
    "two",
  ],
  "type": "FETCH_TAGS_SUCCESS",
}
`;
